include/burner.h

```diff
@@ -4,10 +4,78 @@
 #include "render.h"
 #include "input.h"
 
-typedef void (game_start)(State*, Renderer*);
+typedef void (game_start)(State*, Renderer*, void**);
 typedef game_start* (game_start_loader)();
 
-typedef Mat4 (game_update)(State*, Renderer*, Input*, double, double);
+typedef Mat4 (game_update)(State*, Renderer*, Input*, double, double, void*);
 typedef game_update* (game_update_loader)();
 
+typedef void (game_on_reload)(State*, Renderer*, void**);
+typedef game_on_reload* (game_on_reload_loader)();
+
+
+// forward declare these functions so the game code need not include
+// reload getters
+void start(State*, Renderer*, void**);
+Mat4 update(
+    State*,
+    Renderer*,
+    Input*,
+    double,
+    double,
+    void*
+);
+void on_reload(State*, Renderer*, void**);
+
+#define PASTE2_INTERAL__(x, y) x##y
+#define PASTE_INTERNAL__(x, y) PASTE2_INTERAL__(x, y)
+#include "../out/version.h"
+
+#if defined(DEVELOPMENT_MODE)
+    #pragma message("DEV MODE")
+    #define VERSIONED_PROC(name) PASTE_INTERNAL__(name, LIB_VERSION)
+#else
+    #pragma message("RELEASE MODE")
+    #define VERSIONED_PROC(name) name
+#endif
+
+#if defined(GAME_CODE) && !defined(RELEASE_MODE)
+
+#ifdef __cplusplus
```

Comment:
Will do. The idea here is that I wanted the user not to have to worry about those function callbacks being defined. Those should be provided in the engine. However, the file is included multiple times. To avoid compiler (or linker) errors, I pass in a GAME_CODE flag only to one build step (somewhere in the bash script) that allows the callbacks to be defined. There may be a cleaner way.